app/src/shared/components/TextField/NumberField/index.jsx

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import React, { useCallback, useRef } from 'react'
+import cx from 'classnames'
+
+import SvgIcon from '$shared/components/SvgIcon'
+
+import styles from '../textField.pcss'
+
+type Props = {
+    className?: string,
+    onAutoComplete?: (boolean) => void,
+    type?: string,
+    min?: number,
+    max?: number,
+    step?: number,
+    hideButtons?: boolean,
+}
+
+const NumberField = ({ className, onAutoComplete, hideButtons, ...props }: Props) => {
+    const inputRef = useRef(null)
```

Comment:
Why using a ref? Can't we keep track on the internal state of element's `value` via `useState`?

I found the up/down buttons act inconsistently – something that's definitely related to the ref, see:

![Dec-01-2019 15-44-26](https://user-images.githubusercontent.com/320066/69918218-cd821080-146f-11ea-8ab5-7ef69c002471.gif)

Looks like the value is kept somewhere, but doesn't always stay up-to-date. To reproduce just play around with the field, focus the field, click up and down, go nuts. ;)

Let's make it work nicely!